crescent/ext/tasks/loop.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, Sequence
+
+from crescent.ext.tasks.task import TaskCallbackT, _link_task, _Task
+from crescent.internal import MetaStruct
+
+__all__: Sequence[str] = ("loop",)
+
+
+class _Loop(_Task):
+    def __init__(self, callback: TaskCallbackT, *, hours: int, minutes: int, seconds: int) -> None:
+        self.timedelta = hours * 3600 + minutes * 60 + seconds
+        self.first_loop = True
+
+        super().__init__(callback)
+
+    def time_to_next(self) -> float:
+        if self.first_loop:
+            self.first_loop = False
+            return 0
```

Comment:
This logic breaks the first loop if the user manually calls `time_to_next` (which should be expected since time_to_next isn't a private function).

I personally think this should remain a public function, and just have the logic for setting the first_loop to False handled elsewhere.

Reply:
I changed this so it works. Also changed privacy on some other functions so its clear you can do whatever with this one but not the internal only ones.